### DISCUSSION TITLE
Computing alternate solutions is annoying

### DISCUSSION THREAD

**Discussion by erik-brangs** [eclipse-equinox/p2#340] [General]
In my experience, whenever Eclipse fails to find a solution when updating and has to resort to "Compute alternate solutions", the user experience is pretty bad. Finding alternate solutions takes a long time. There is already #309 about this.

The last time when I encountered "Compute alternate solutions", the dialog had focus and I could not minimize it to continue working in the meantime. Is this still the case? And if it is, could this be changed or is it unavoidable?

**Comment by merks** [comment]
I'm not sure anyone will consider this a priority on which to work.  There's just so much to do!  Perhaps you consider it a priority and would like to contribute toward your personal priorities:

https://github.com/eclipse-equinox/.github/blob/main/CONTRIBUTING.md

The problem with a non-modal dialog is that one has to be careful that one don't start yet another update or install that will conflict with the on-going operation, i.e., it's not entirely trivial just to make the dialog non-modal.

**Reply by mickaelistria** [reply]
> The problem with a non-modal dialog is that one has to be careful that one don't start yet another update or install that will conflict with the on-going operation, i.e., it's not entirely trivial just to make the dialog non-modal.

I think p2 has a "profile version" and a "profile lock" whenever an installation is ongoing to prevent from this kind of issues already.

**Comment by erik-brangs** [comment]
Thank you. It sounds like this might be be feasible to do. I suppose I'll wait and see if it continues to annoy me. It might annoy me enough to do something about it.

With the plugins that I have installed, the "Compute alternate solutions" took several minutes on a few separate occasions.

**Reply by mickaelistria** [reply]
> The last time when I encountered "Compute alternate solutions", the dialog had focus and I could not minimize it to continue working in the meantime.

Please report a specific issue for this.

> Is this still the case?

The best way for you to know is to try it ;)

> And if it is, could this be changed or is it unavoidable?

It is most likely avoidable.

**Reply by erik-brangs** [reply]
I've opened #341.
